compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1095,14 +1095,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     ty::PredicateKind::Clause(ty::Clause::RegionOutlives(..))
                     | ty::PredicateKind::Clause(ty::Clause::Projection(..))
                     | ty::PredicateKind::Clause(ty::Clause::TypeOutlives(..)) => {
-                        let predicate = self.resolve_vars_if_possible(obligation.predicate);
-                        struct_span_err!(
-                            self.tcx.sess,
-                            span,
-                            E0280,
-                            "the requirement `{}` is not satisfied",
-                            predicate
-                        )
+                        span_bug!(span, "clause requirement gave wrong error: `{:?}`", obligation)
                     }
```

Comment:
I think it would be helpful to split this into `RegionOutlives` and `TypeOutlives` with a message of "outlives clauses should not error outside borrowck" and `Projection` with a message of "projection clauses should be implied from elsewhere"